DynamicTablesPkg/Include/ArchCommonNameSpaceObjects.h

```diff
@@ -812,6 +818,20 @@ typedef struct CmArchCommonStaInfo {
   UINT32    DeviceStatus;
 } CM_ARCH_COMMON_STA_INFO;
 
+/**
+  A structure that describes the SLIT (System Locality Information Table)
+    information.
+
+    ID: EArchCommonObjSlitInfo
+*/
+typedef struct CmArchCommonSlitInfo {
+  /// The number of localities supported by the SLIT table.
+  UINT8    NumberOfSystemLocalities;
+
+  /// The distance between each pair of localities.
+  UINT8    Entity[EARCH_COMMON_SLIT_LOCALITY_MAX * EARCH_COMMON_SLIT_LOCALITY_MAX];

```

Comment:
Theoretically there should be no reason to limit the number of locality domains.
Also, there should be a rework of how ProximityDomains are generated. Ideally, the developer should not have to keep track of the IDs.